Scripts/Runtime/Entities/Lifecycle/AbstractEntityLifecycleStatusSystem.cs

```diff
@@ -0,0 +1,303 @@
+            m_EntityLifecycleStatus.Add(entityLifecycleStatus);
+            return entityLifecycleStatus;
+        }
+
+        protected override void OnUpdate()
+        {
+            JobHandle dependsOn = Dependency;
+            dependsOn = m_WorldEntityState.AcquireCreatedAndDestroyedEntities(
+                dependsOn,
+                out NativeArray<Entity>.ReadOnly createdEntities,
+                out NativeArray<Entity>.ReadOnly destroyedEntities,
+                m_ShouldCleanupEntitiesCountAsDepartures);
+
+            //If nothing was created or destroyed we can early exit
+            if (createdEntities.Length == 0
+                && destroyedEntities.Length == 0)
+            {
+                m_WorldEntityState.ReleaseCreatedAndDestroyedEntities(dependsOn);
+                Dependency = dependsOn;
+                return;
+            }
+
+            Logger.Debug($"{World} | {UnityEngine.Time.frameCount} - Created: {createdEntities.Length} - Destroyed: {destroyedEntities.Length}");
+            dependsOn = UpdateAsync(
+                dependsOn,
+                ref destroyedEntities);
+
+            m_WorldEntityState.ReleaseCreatedAndDestroyedEntities(dependsOn);
+            Dependency = dependsOn;
+        }
+
+        private JobHandle UpdateAsync(
+            JobHandle dependsOn,
+            ref NativeArray<Entity>.ReadOnly destroyedEntities)
+        {
+            for (int i = 0; i < m_EntityLifecycleStatus.Count; ++i)
+            {
+                m_Dependencies[i] = m_EntityLifecycleStatus[i]
+                    .UpdateAsync(
+                        dependsOn,
+                        ref destroyedEntities);
+            }
+            return JobHandle.CombineDependencies(m_Dependencies);
+        }
+
+        //*************************************************************************************************************
+        // WORLD ENTITY STATE
+        //*************************************************************************************************************
+        
+        private class WorldEntityState
+        {
+            private static readonly Dictionary<World, WorldEntityState> s_WorldEntityStates = new Dictionary<World, WorldEntityState>();
+
+            [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+            private static void Init()
+            {
+                s_WorldEntityStates.Clear();
+            }
+
+            public static WorldEntityState GetOrCreate(World world, AbstractEntityLifecycleStatusSystem lifecycleStatusSystem)
+            {
+                if (!s_WorldEntityStates.TryGetValue(world, out WorldEntityState worldEntityState))
+                {
+                    worldEntityState = new WorldEntityState(world);
+                    s_WorldEntityStates.Add(world, worldEntityState);
+                }
+                worldEntityState.AddLifecycleStatusSystem(lifecycleStatusSystem);
```

Comment:
This should be a separate public method or described in the method name (Ex: `GetOrCreateAndAddLifecycleSystem`)

In the spirit of single responsibility methods my preference is to make `AddLifecycleStatusSystem` a separate public method that needs to be called.

Reply:
Ok done